modules/GenomicsEnglandPanelApp.py

```diff
@@ -110,53 +154,113 @@ def publication_from_symbol(symbol, response):
                 publication = gene["publications"]
                 return publication
 
+
     @staticmethod
-    def clean_dataframe(dataframe):
+    def cleanDataframe(dataframe):
         '''
         Args:
-            dataframe (pd.DataFrame): Initial .tsv data converted to a Pandas dataframe
+            dataframe (pyspark.DataFrame): Initial .tsv data
         Returns:
-            dataframe (pd.DataFrame): Original dataframe cleaned
+            dataframe (pyspark.DataFrame): Transformed initial dataframe
         '''
-        # NaNs and "No OMIM phenotype" in "Phenotypes" column --> Assignment of Pannel Name
-        dataframe.loc[(dataframe["Phenotypes"] == "No OMIM phenotype") | (dataframe["Phenotypes"].isna()), "Phenotypes"] = dataframe["Panel Name"]
-
-        # Handling multiple phenotypes column: Phenotypes exploded and phenotype_list with the original string
-        dataframe["phenotype_list"] = dataframe["Phenotypes"]
-        dataframe["Phenotype"] = dataframe["Phenotypes"].apply(lambda X: X.split(";"))
-        dataframe = dataframe.explode("Phenotype")
-
-        # Extracting and cleaning the OMIM codes: 
-        #   removal of the OMIM codes in the Phenotypes column and the inclusion of these in a Codes column
-        #   deleting special characters
-        dataframe["OMIM_code"] = dataframe["Phenotype"].str.extract(r"(\d{6})")
-        dataframe["Phenotype"] = dataframe["Phenotype"].replace("(\d{6})", "", regex=True)
+    
+        dataframe = (dataframe
+                        .withColumn(
+                            # NaNs and "No OMIM phenotype" in "Phenotypes" column --> Assignment of Panel Name
+                            "Phenotypes",
+                            when(col("Phenotypes") == "No OMIM phenotype",
+                                col("Panel Name"))
+                            .when(col("Phenotypes").isNull(),
+                                col("Panel Name"))
+                            .otherwise(col("Phenotypes"))
+                        )
+                        # cohortPhenotypes --> array of the original string separated by phenotypes 
+                        .withColumn(
+                            "cohortPhenotypes",                          
+                            split(col("Phenotypes"), ";")
```

Comment:
Ocassionally the generated evidence fail because this array is not unique eg:

```
{
  "datasourceId": "genomics_england",
  "datatypeId": "genetic_literature",
  "confidence": "green",
  "diseaseFromSource": "Optic atrophy 1",
  "diseaseFromSourceId": "165500",
  "diseaseFromSourceMappedId": "Orphanet_98673",
  "cohortPhenotypes": [
    "Disorders of mitochondrial DNA maintenance and integrity",
    "Optic atrophy 1, 165500",
    "{Glaucoma, normal tension, susceptibility to}, 606657",
    "Optic atrophy plus syndrome, 125250",
    "Mitochondrial DNA Depletion Syndrome",
    "Progressive External Ophthalmoplegia with Mitochondrial DNA Deletions",
    "Optic atrophy 1, 165500"
  ],
  "targetFromSourceId": "OPA1",
  "allelicRequirements": [
    "BOTH monoallelic and biallelic (but BIALLELIC mutations cause a more SEVERE disease form), autosomal or pseudoautosomal"
  ],
  "studyId": "112",
  "studyOverview": "Mitochondrial disorders"
}
```

Reply:
Thanks for pointing this out. Fixed.